### ISSUE TITLE
Proxy detected on your Now OSS Plan

### ISSUE THREAD

**Issue by webmasterslava** [coin-hive-proxy-party/coin-hive-proxy-party.github.io#3]
Just got a message from Zeit.co

> We have observed you deployed a proxy to the Open Source plan on Now. The OSS plan is not intended to serve as a traffic relay or CDN.
> 
> If you don't upgrade to a paid plan, we will shut down your deployments shortly.
> 
> Thank you for using Now!

Is it possible to do something to keep working using free OSS plan?

**Comment by cazala** [comment]
Mmh, technically this is not a proxy, but a gateway.. I changed the name of the package so it doesn't say proxy.

But other than that, I dunno..

You could try to deploy a new instance with the new package, or take that package.json and deploy it somewhere else like heroku..

Their ToS didn't say anything about gateways/proxies/relays at all, but I guess they do have the right to take down whatever they want on their free plans, so... ¯\\\_(ツ)_/¯

**Comment by janowitz** [comment]
As already noted on reddit I got the very same Email, but I think it's a traffic issue, since I caused 226GB of traffic within few hours.
How can we make sure we use their OSS for the websocket only and deliver all static files with proper cache headers over some other CDN / Github?

**Comment by cazala** [comment]
I'll need to make a slight change in order to make this possible (right now the miner uses the same URL for the proxy and for the assets, i need to make these two separate urls). I'll post it here once it's done, then you will need to deploy a new instance in order to use this.

**Comment by janowitz** [comment]
Great, thanks for your effort!
I noticed, the JS is obfuscated, what method are you using for that?

**Comment by cazala** [comment]
Just ascii to base64

Sent from my iPhone

On Oct 26, 2017, at 11:35 AM, janowitz <notifications@github.com<mailto:notifications@github.com>> wrote:


Great, thanks for your effort!
I noticed, the JS is obfuscated, what method are you using for that?

**Comment by janowitz** [comment]
I wonder if it would be possible to clean the script up a little bit and get rid of all the hard coded coinhive stuff... Unfortunately JavaScript is not my language and it's hard to follow for me the exact stuff going on there.

**Comment by cazala** [comment]
what hardcoded coinhive stuff?

Sent from my iPhone

On Oct 26, 2017, at 5:51 PM, janowitz <notifications@github.com<mailto:notifications@github.com>> wrote:


I wonder if it would be possible to clean the script up a little bit and get rid of all the hard coded coinhive stuff... Unfortunately JavaScript is not my language and it's hard to follow for me the exact stuff going on there.

**Comment by janowitz** [comment]
Hardcoded proxy URLs which bloat the file quite a lot:
`
  WEBSOCKET_SHARDS: [
    [
      "wss://ws001.coinhive.com/proxy",
      "wss://ws002.coinhive.com/proxy",
      "wss://ws003.coinhive.com/proxy",
      "wss://ws004.coinhive.com/proxy",
      "wss://ws005.coinhive.com/proxy",
      "wss://ws006.coinhive.com/proxy",
      "wss://ws007.coinhive.com/proxy",
      "wss://ws029.coinhive.com/proxy"
    ],
`

**Comment by janowitz** [comment]
When it comes to the original topic, Zeit just sent me the exact same email like before few hours after deployment on a fresh account but not using it at all. So they are probably actively scanning for it.

**Comment by phr34k0** [comment]
Yes it's pretty useless anything with a CoinHive name in the script gets detected. That was why I said it the previous git (https://github.com/cazala/coin-hive-proxy) self host with ability to rename and deploy was a better option than that of (https://coinhive-proxy.party)

https://phr34k0.github.io/ifixit.pw2/ is detected by adguard. Bet it's detected by anti viruses too.

**Comment by phr34k0** [comment]
![capture](https://user-images.githubusercontent.com/13272249/32130864-f7598994-bbd2-11e7-9bb3-a2cde49827cc.JPG)

**Comment by coin-hive-proxy-party** [comment]
This project has been deprecated, if you are still interested please check [CoinHive Stratum](https://github.com/cazala/coin-hive-stratum/wiki/Deploy-to-now.sh-and-GitHub-Pages).
